nibabel/cifti2/cifti2.py

```diff
@@ -1490,6 +1503,33 @@ def get_data_dtype(self):
     def set_data_dtype(self, dtype):
         self._nifti_header.set_data_dtype(dtype)
 
+    def to_filename(self, filename, infer_intent=False):
+        """
+        Ensures NIfTI header intent code is set prior to saving.
+
+        Parameters
+        ----------
+        infer_intent : boolean, optional
+            If ``True``, attempt to infer and set intent code based on filename suffix.
+        """
+        header = self._nifti_header
+        if infer_intent:
+            # try to infer intent code based on filename suffix
+            intent = _infer_intent_from_filename(filename)
+            if intent is not None:
+                header.set_intent(intent)
+        # if intent code is not set, default to unknown
+        if header.get_intent()[0] == 'none':
+            header.set_intent('NIFTI_INTENT_CONNECTIVITY_UNKNOWN')
+        super().to_filename(filename)
+
+
+def _infer_intent_from_filename(filename):
+    """Parses output filename for common suffixes and fetches corresponding intent code"""
+    from pathlib import Path
+    ext = Path(filename).suffixes[0]
```

Comment:
This might be more reliable (though require an `except IndexError` case):

```suggestion
    ext = Path(filename).suffixes[-2]
```